modules/images/webp-uploads/load.php

```diff
@@ -340,3 +340,112 @@ function webp_uploads_wp_get_missing_image_subsizes( $missing_sizes, $image_meta
 }
 
 add_filter( 'wp_get_missing_image_subsizes', 'webp_uploads_wp_get_missing_image_subsizes', 10, 3 );
+
+/**
+ * Filters on `the_content` to update the references for supported mime of images into the
+ * `webp_uploads_preferred_mime_type()` for the most part `image/webp` if the current
+ * attachment contains the targeted mime type.
+ *
+ * @since n.e.x.t
+ *
+ * @see the_content
+ *
+ * @param string $content The content of the current post.
+ * @return string The content with the updated references to the images.
+ */
+function webp_uploads_update_image_references( $content ) {
+	// This content does not have any tag on it, move forward.
+	if ( ! preg_match_all( '/<(img)\s[^>]+>/', $content, $img_tags, PREG_SET_ORDER ) ) {
+		return $content;
+	}
+
+	$images = array();
+	foreach ( $img_tags as list( $img ) ) {
+		// Find the ID of each image by the class.
+		if ( ! preg_match( '/wp-image-([\d]+)/i', $img, $class_name ) ) {
+			continue;
+		}
+
+		if ( empty( $class_name ) ) {
+			continue;
+		}
+
+		// Make sure we use the last item on the list of matches.
+		$attachment_id = (int) end( $class_name );
+
+		if ( ! $attachment_id ) {
+			continue;
+		}
+
+		// Create an array if attachment_id has not been added to the images array.
+		if ( empty( $images[ $attachment_id ] ) ) {
+			$images[ $attachment_id ] = array();
+		}
+
+		$target_extensions = array(
+			'jpeg',
+			'jpg',
+		);
+
+		// Creates a regular extension to find all the URLS with the provided extension for img tag.
+		preg_match_all( '/[^\s"]+\.(?:' . implode( '|', $target_extensions ) . ')/i', $img, $matches );
+
+		$urls = empty( $matches ) ? array() : reset( $matches );
+
+		foreach ( $urls as $url ) {
+			$images[ $attachment_id ][ $url ] = true;
+		}
+	}
+
+	// TODO: Add a filterable option to change the selected mime type. See https://github.com/WordPress/performance/issues/187.
+	$target_mime = 'image/webp';
+	$replacement = array();
+	foreach ( $images as $attachment_id => $urls ) {
```

Comment:
> Split it into two functions. One that does the content image parsing (which we won't need later in core) and the other to modify an individual image, similar to e.g. wp_img_tag_add_srcset_and_sizes_attr().

Agree on this one.

> The variable should become a simple map like $image => $attachment_id (image HTML tag and attachment ID), the exact same how WP core's wp_filter_content_tags() has it. This way we're already building a compatible implementation which we can later just plug into core.

Actually on this one having a hash map makes more sense and is more optimal due if an image is included 10 times on the content the logic to replace the same image would take place 10 times instead of a single time if we are using a hash map as we currently have this logic will ensure every single iteration is unique based on the image ID, Using the core version currently does not consider this scenario. 

> For example, the second function could be something like webp_uploads_img_tag_choose_mime( $image, $context, $attachment_id ), and it would return the modified $image tag. All the logic you have below here should then be moved into that function. The above preg_match_all to find all URLs would also need to be included there, instead of having it above, which isn't really necessary.

Agree on that one.

Reply:
> Actually on this one having a hash map makes more sense and is more optimal due if an image is included 10 times on the content the logic to replace the same image would take place 10 times instead of a single time if we are using a hash map as we currently have this logic will ensure every single iteration is unique based on the image ID, Using the core version currently does not consider this scenario.

Can you clarify how you mean that? If an image is included 10 times in the same content, with my suggestion above (`$image => $attachment_id` map) the logic would still only be run once, unless the same image was included in 10 different versions (e.g. if someone included it as `medium`, `large`, `full`, `post-thumbnail` etc., all in a single post), which is an edge-case.

Good point on the URL lookup though, it makes sense to find a solution where we don't need to look up the replacement for the same URL multiple times. But still, let's decouple that storage from the `$image => $attachment_id` map, which is needed for seamless integration in core. We can still store the URLs separately, in a way that the lookup doesn't need to happen more than once per URL.

Reply:
> Can you clarify how you mean that? If an image is included 10 times in the same content, with my suggestion above ($image => $attachment_id map) the logic would still only be run once, unless the same image was included in 10 different versions (e.g. if someone included it as medium, large, full, post-thumbnail etc., all in a single post), which is an edge-case.

The fact that we need to look up for URLS besides just `<img>` tags required a way to uniquely identity each URL to avoid duplicate work, I've updated the code based on your feedback as I think it makes more sense as you describe now that the logic is placed in 2 places.

I've committed the `context` and `attachment_id` on the second function as those were never used I feel like we can add those once the migration in core lands due it would make sense in that context but not so much under our current filter, let me know otherwise.